log/wrapper.go

```diff
@@ -272,6 +273,27 @@ func ErrorWithSentryWrapper() Wrapper {
 	}
 }
 
+// PrometheusCounterWrapper returns a Wrapper implementation that increases
+// counter by 1 then calls delegate to log the message.
+//
+// Please note that it's not possible to deserialize this Wrapper directly from
+// yaml, so you usually need to override it in your main function, after
+// baseplate.ParseConfigYAML call, for example:
+//
+//     if err := baseplate.ParseConfigYAML(&cfg); err != nil {
+//       log.Fatal(err)
+//     }
+//     cfg.Config.Tracing.Logger = log.PrometheusCounterWrapper(
+//       cfg.Config.Tracing.Logger,       // delegate
+//       tracingFailureCounter.With(nil), // counter
```

Comment:
why the `.With(nil)`?

Reply:
In a typical prometheus counter, the global variable you defined is a [`CounterVec`](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#CounterVec), and [`With`](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#CounterVec.With) gives you an actual [`Counter`](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#Counter).

The args to `With` is the label values. For things used globally as a log wrapper, you usually don't really have any labels defined for the counter, thus the `nil`.

Reply:
would it make sense to just do `counter = counter.With(nil)` in PrometheusCounterWrapper before returning the function?

Reply:
actually let me expand the example a little bit.